examples/all-clusters-app/all-clusters-common/all-clusters-app.matter

```diff
@@ -1167,6 +1172,21 @@ server cluster PowerSource = 47 {
     kReplaceable = 0x8;
   }
 
+  struct BatChargeFaultChangeType {
```

Comment:
They are in the XML because someone messed up while defining the events at some point and defined "payload structs" for them instead of defining the events themselves to have fields.

These structs are not used by anything at this point, but removing them completely breaks backwards compat for some APIs that are exposing them at the moment, which is why they have not been removed yet.

Reply:
What APIs are exposing this currently?

What I am wondering is that previous codegen for IDL did not generate them and that seemed to look better because we do not generate unused structures. I thought this was a feature not a bug. Will not break codegen logic, it just will codegen more than needed.